rally-domain/src/main/groovy/yakworks/rally/mail/EmailUtils.groovy

```diff
@@ -24,21 +24,27 @@ class EmailUtils {
      * Examples: <br>
      * "joe@email.com" - simple single email <br>
      * "Account Services <tesla@greenbill.io>, "Galt, John" <galt@yak.works>, joe@email.com" - 3 valid emails
-     * @return true is valid
-     * @throw ThrowableProblem with code:validation.problem
+     *
+     * @throws ThrowableProblem if validation fails
      */
-    static boolean validateEmail(String emails) {
-        boolean isValid = false
+    static void validateEmail(String emails) {
+        if(!emails?.trim()) {
+            throw Problem.of("validation.problem").detail("Empty email").toException()
+        }
         try {
             InternetAddress[] addys = InternetAddress.parse(emails)
-            addys.each{
-                it.validate()
+            for (InternetAddress addr : addys) {
+                try {
+                    addr.validate()
+                } catch (AddressException e) {
+                    throw Problem.of("validation.problem")
+                        .payload(addr)
+                        .detail("Invalid email address [$addr], " + e.message).toException()
+                }
             }
-            isValid = true;
         } catch (AddressException e) {
-            throw Problem.of("validation.problem").detail(e.message).toException()
+            throw Problem.of("validation.problem").payload(emails).detail(e.message).toException()
```

Comment:
what this one for? the `InternetAddress.parse(emails)`?

Reply:
Yes, parse can throw error too